### PR TITLE
document required invocations to build docs

### PR DESCRIPTION
**Pull request by lubegasimon**
Addresses #460

### CODE REVIEW

File: src/odoc/system-integrations.mld [hunk 1]
```diff
@@ -17,4 +17,197 @@ You can use the {{:https://reasonml-community.github.io/bsdoc}[bsdoc]} npm packa
...
+Additionally, it was noted that modules are treated as directories in the output, with
+an [index.html] page describing the module itselt, and further subdirectories related to
+submodules. These concepts have been unified in [odoc v2.0.0] that comes with
+{e parent\child specification}; thus allowing mld files to have {i children} in a similar way
+to modules having subdirectories.
+
+{3 Parent/child specification}
+This parent/child concept will allow more flexible output support – for example, per-library
+documentation.
+This will be seen in {{:https://docs.ocaml.org}[docs.ocaml.org]}.
```
Comment (lubegasimon) [inline src/odoc/system-integrations.mld:51]: `docs.ocaml.org` site isn’t up at the moment. That acted as a placeholder.

File: src/odoc/system-integrations.mld [hunk 2]
```diff
@@ -17,4 +17,197 @@ You can use the {{:https://reasonml-community.github.io/bsdoc}[bsdoc]} npm packa
...
+
+For example, let's consider [John] whose is a father to [Doe] and [Mark]. [Doe] has
+a child called [Max], and module [foo], whereas [Mark] has no child. That is to say,
+[john.mld], [doe.mld], [mark.mld], [max.mld], [foo.ml] respectively. For instance;
+
+[john.mld]
+{[
+  {0 About John}
+
+  I'm John the father to {{!page-doe}Doe} and {{!page-mark}Mark}.
```
Comment (lubegasimon) [inline src/odoc/system-integrations.mld:79]: > {{!page-doe}Doe}

I’m not sure if it makes sense to have bare syntax in this document.

File: src/odoc/system-integrations.mld [hunk 3]
```diff
@@ -17,4 +17,197 @@ You can use the {{:https://reasonml-community.github.io/bsdoc}[bsdoc]} npm packa
...
+underneath that package specified by [--pkg]/[--package].
+
+For example,
+
+[odoc compile -I . --pkg <package> -o <file>.odoc <file>.cmti]
+
+where [<file>.cmti] is the result of compiling [<file>.mli] using [OCaml] compiler [ocamlc].
+
+Additionally, it was noted that modules are treated as directories in the output, with
+an [index.html] page describing the module itselt, and further subdirectories related to
```
Comment (jonludlam) [inline src/odoc/system-integrations.mld:43]: *itself

File: src/odoc/system-integrations.mld [hunk 4]
```diff
@@ -17,4 +17,197 @@ You can use the {{:https://reasonml-community.github.io/bsdoc}[bsdoc]} npm packa
...
+
+{3 Parent/child specification}
+This parent/child concept will allow more flexible output support – for example, per-library
+documentation.
+This will be seen in {{:https://docs.ocaml.org}[docs.ocaml.org]}.
+
+The rules are;
+- Mld files may or may not have a parent mld.
+- Compilation units must have a parent mld.
+- The parent mld file must be compiled before any of it's children, and it's children
```
Comment (jonludlam) [inline src/odoc/system-integrations.mld:56]: *its

File: src/odoc/system-integrations.mld [hunk 5]
```diff
@@ -17,4 +17,197 @@ You can use the {{:https://reasonml-community.github.io/bsdoc}[bsdoc]} npm packa
...
+  must be specified at compilation time of the parent
+- The output paths of mlds and compilation units are subdirectories of their parent's
+  output directory.
+- The output directory of an mld file [x] with children is [<parent_output_directory>
+  /x], and it's file name is [index.html]. That is to say, [<parent_output_directory>
```
Comment (jonludlam) [inline src/odoc/system-integrations.mld:61]: *its

File: src/odoc/system-integrations.mld [hunk 6]
```diff
@@ -17,4 +17,197 @@ You can use the {{:https://reasonml-community.github.io/bsdoc}[bsdoc]} npm packa
...
+  {0 About John}
+
+  I'm John the father to {{!page-doe}Doe} and {{!page-mark}Mark}.
+]}
+
+[doe.mld]
+{[
+  {0 About Doe}
+
+  I'm Deo, the;
```
Comment (jonludlam) [inline src/odoc/system-integrations.mld:86]: Doe or Deo?

**Comment by jonludlam** [comment]
has this been superseded by #683 ?

**Comment by jonludlam** [comment]
I believe this _has_ been superseded.

**Comment by lubegasimon** [comment]
> I believe this _has_ been superseded.

Right, thanks!

**Pull request closed without merging**